contracts/twap/CumulativeTwap.sol

```diff
@@ -129,11 +147,15 @@ contract CumulativeTwap is BlockContext {
 
         // if the target is chronologically at or after the newest observation, we can early return
         if (observations[currentObservationIndex].timestamp <= targetTimestamp) {
-            return (beforeOrAt, beforeOrAt);
+            // if the observation is the same as the targetTimestamp
+            // atOrAfter doesn't matter
+            // if the observation is less than the targetTimestamp
+            // atOrAfter repesents latest price and timestamp
+            return (beforeOrAt, atOrAfter);
         }
 
         // now, set before to the oldest observation
-        beforeOrAt = observations[(currentObservationIndex + 1) % MAX_OBSERVATION];
```

Comment:
Need some help: why is this remainder removed?

Reply:
Oh i just realised that there's a previous pr to this one: https://github.com/perpetual-protocol/perp-oracle-contract/pull/61
Will take a look at that and get back to this one

Reply:
you're right. I merge it with a wrong logic accidently.